mcp/tools/tools_coder.go

```diff
@@ -0,0 +1,381 @@
+package mcptools
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"io"
+	"strings"
+
+	"github.com/google/uuid"
+	"github.com/mark3labs/mcp-go/mcp"
+	mcpserver "github.com/mark3labs/mcp-go/server"
+	"golang.org/x/xerrors"
+
+	"cdr.dev/slog"
+	"github.com/coder/coder/v2/coderd/util/ptr"
+	"github.com/coder/coder/v2/codersdk"
+	"github.com/coder/coder/v2/codersdk/workspacesdk"
+)
+
+// Example payload:
+// {"jsonrpc":"2.0","id":1,"method":"tools/call", "params": {"name": "coder_report_task", "arguments": {"summary": "I'm working on the login page.", "link": "https://github.com/coder/coder/pull/1234", "emoji": "🔍", "done": false}}}
+func handleCoderReportTask(deps ToolDeps) mcpserver.ToolHandlerFunc {
+	return func(ctx context.Context, request mcp.CallToolRequest) (*mcp.CallToolResult, error) {
+		if deps.Client == nil {
+			return nil, xerrors.New("developer error: client is required")
+		}
+
+		args := request.Params.Arguments
+
+		summary, ok := args["summary"].(string)
+		if !ok {
+			return nil, xerrors.New("summary is required")
+		}
+
+		link, ok := args["link"].(string)
+		if !ok {
+			return nil, xerrors.New("link is required")
+		}
+
+		emoji, ok := args["emoji"].(string)
+		if !ok {
+			return nil, xerrors.New("emoji is required")
+		}
+
+		done, ok := args["done"].(bool)
+		if !ok {
+			return nil, xerrors.New("done is required")
+		}
+
+		// TODO: Waiting on support for tasks.
+		deps.Logger.Info(ctx, "report task tool called", slog.F("summary", summary), slog.F("link", link), slog.F("done", done), slog.F("emoji", emoji))
+		/*
```

Comment:
dead code on purpose?

Reply:
We can remove this if you prefer, but I was anticipating this being ready sooner.